### PR TITLE
Feature/task run metadata struct

### PR DESCRIPTION
**Pull request by RiddhiBagadiaa**
# Description

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: scheduling/task_run.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	}
+	t.Status = temp.Status
+
+	if temp.StartTime.IsZero() {
+		return fmt.Errorf("task run metadata is missing Start Time")
+	}
+	t.StartTime = temp.StartTime
+
+	if temp.EndTime.IsZero() {
+		return fmt.Errorf("task run metadata is missing End Time")
```
Comment (ahmadnazeri) [inline scheduling/task_run.go:129]: for errors, we might have to leverage the errors package if it gets merged.

File: scheduling/task_run.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
+package scheduling
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type TaskRunId int32
+type TaskId int32 // TODO: This datatype already exists in rasks.go, so when we merge, we should use the same?
```
Comment (sdreyer) [inline scheduling/task_run.go:10]: well resolve this conflict when we get the other merged

File: scheduling/task_run.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+type Status string
+
+const (
+	Success Status = "SUCCESS"
```
Comment (sdreyer) [inline scheduling/task_run.go:14]: Would like a RUNNING status too

File: scheduling/task_run.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+	return t.TriggerType
+}
+
+func (t DummyTrigger) Name() string {
+	return t.TriggerName
+}
+
+type TaskRunMetadata struct {
+	ID        TaskRunId `json:"id"`
+	RunId     TaskId    `json:"runId"`
```
Comment (sdreyer) [inline scheduling/task_run.go:60]: This should be TaskID. The ID above is already the Run ID

Comment (sdreyer) [reply scheduling/task_run.go:60]: change to run_id

File: scheduling/task_run.go [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+	Name      string    `json:"name"`
+	Trigger   Trigger   `json:"trigger"`
+	Status    Status    `json:"status"`
+	StartTime time.Time `json:"startTime"`
```
Comment (sdreyer) [inline scheduling/task_run.go:64]: change to start_time and end_time

File: scheduling/task_run.go [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+// func (t *TaskRunMetadata) Error() string
+
+// Formatting
+func (t *TaskRunMetadata) ToJSON() ([]byte, error) {
+	return json.Marshal(t)
+}
+func (t *TaskRunMetadata) FromJSON(data []byte) error {
+	type tempConfig struct {
+		ID        TaskRunId       `json:"id"`
+		RunId     TaskId          `json:"runId"`
```
Comment (sdreyer) [inline scheduling/task_run.go:88]: fix json keys from camel case to snake case

File: scheduling/task_run.go (hunk 1 above)
Comment (sdreyer) [reply scheduling/task_run.go:129]: Yeah, im leaving the errors as-is for now, but once the errors package gets merged we need to swap these out

File: scheduling/task_run_test.go [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+				ID:    1,
+				RunId: 12,
+				Name:  "oneoff_taskrun",
+				Trigger: OneOffTrigger{
+					TriggerName: "name1",
+					TriggerType: oneOffTrigger,
+				},
+				Status:    Pending,
+				StartTime: time.Now().Truncate(0),
+				EndTime:   time.Now().Truncate(0),
```
Comment (sdreyer) [inline scheduling/task_run_test.go:27]: try adding .UTC() to the end of this

File: scheduling/task_run.go [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+// func (t *TaskRunMetadata) Name() string
+// func (t *TaskRunMetadata) Trigger() Trigger
+// func (t *TaskRunMetadata) Status() Status // Enum
+// func (t *TaskRunMetadata) StartTime() Time
+// func (t *TaskRunMetadata) EndTime() Time
+// func (t *TaskRunMetadata) Logs() []string
+// func (t *TaskRunMetadata) Error() string
+
+// Formatting
+func (t *TaskRunMetadata) ToJSON() ([]byte, error) {
```
Comment (aolfat) [inline scheduling/task_run.go:82]: Are we not going to use the traditional marshalling method names we talked about?

File: scheduling/task_run.go [hunk 9]
```diff
@@ -0,0 +1,160 @@
+package scheduling
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
```
Comment (aolfat) [inline scheduling/task_run.go:8]: A comment at the top of the file describing what a task is would be really useful

File: scheduling/task_run.go (hunk 5 above)
Comment (aolfat) [reply scheduling/task_run.go:64]: i would probably do started_at and ended_at and provide a comment on what type of timing it is. UTC?

**Pull request closed without merging**